### PR TITLE
chore(debugger): support for all metric kinds with value expression

### PR DESCRIPTION
**Pull request by shatzi**
## Description

Effort to make feature parity with dd-trace-java metric implementation. 
To gain parity python debugger need to support:
1. GAUGE and HISTOGRAM metrics. 
2. Be able to evaluate the metric value expression
3. Support Method Metric Probe with duration

We also decided to support DISTRIBUTION metric support as well. 
This PR address the two main features. future PR would add support for function level Metric Probe

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: ddtrace/debugging/_probe/remoteconfig.py [hunk 1]
```diff
@@ -116,6 +116,7 @@ def probe(_id, _type, attribs):
             line=int(attribs["where"]["lines"][0]),
             name=attribs["metricName"],
             kind=attribs["kind"],
+            value=_compile_condition(attribs.get("value")),
```
Comment (P403n1x87) [inline ddtrace/debugging/_probe/remoteconfig.py:119]: We should probably rename this to `_compile_expression` at this point

Comment (shatzi) [reply ddtrace/debugging/_probe/remoteconfig.py:119]: renamed

Comment (P403n1x87) [reply ddtrace/debugging/_probe/remoteconfig.py:119]: Just checking that we didn't miss a change here. the diff view is still showing the old name here.

**Review by P403n1x87 — APPROVED** [review]
💪

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**